runtime/dolphin/src/lib.rs

```diff
@@ -264,6 +265,7 @@ impl Contains<Call> for BaseFilter {
             | Call::TechnicalMembership(_)
             | Call::Scheduler(_)
             | Call::Session(_) // User must be able to set their session key when applying for a collator
+            | Call::AuthorInherent(pallet_author_inherent::Call::kick_off_authorship_validation {..}) // executes unsigned on every block
```

Comment:
Are the inherents called before the on_initialize hooks ?

Reply:
Shawn's got the answer [here](https://substrate.stackexchange.com/a/3783/504)
Inherents are called before extrinsics and after `on_initialize` is relsolved.

Reply:
Thanks, that clears it up. 
So it's `pallet 1 on_init` -> `pallet 1 intrinsic` -> `pallet 2 on_init` -> `pallet 2 intrinsic` etc...

Reply:
actually no, it's three distinct phases

1. all `on_init` hooks by hook order, then
2. all inherents by pallet order, then
3. extrinsics

Reply:
Ok thanks for clarifying again.
I really haven't found good documentation on this process, besides reading the code, which I don't have the best entry point for...